### PR TITLE
DrivAerML: 4L/256d+T_max=30 — max-eval-batches=400 (denser eval)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
The baseline uses `--max-eval-batches 200`, which evaluates roughly 200 car configurations per validation pass. Doubling this to 400 gives the optimizer a denser, lower-variance gradient signal on the validation distribution. With only ~500 cars total in DrivAerML, evaluating 200 may miss important edge-case geometries that could guide better learning. Denser evaluation also gives us a more reliable primary metric estimate, which helps us make better merge decisions.

Note: This experiment doesn't change the model or training — it changes evaluation coverage. The goal is to see whether higher-fidelity validation feedback (used as the checkpoint-selection criterion) allows the model to converge to a better optimum.

## Instructions

Run the following command:

```bash
cd target/icml2026 && python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 5e-4 \
  --cosine_t_max 30 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 400 \
  --wandb_group drivaerml-4L256d-training-tricks
```

Key changes from baseline:
- `--max-eval-batches 400` (was 200, now 2x more evaluation coverage)
- `--wandb_group drivaerml-4L256d-training-tricks` for grouping

All OOM-prevention flags are preserved: `--batch-size 1`, `--drivaerml-train-surface-points 50000`, `--drivaerml-eval-surface-points 50000`, `--max-train-batches 394`, `--model-heads 4`.

Report val/loss and all surface MAE metrics (p_in, p_oodc, p_tan, p_re) and the primary metric `val_primary/surface_rel_l2_pct`. Compare against baseline of **12.70%**.

## Baseline

Current best: PR #2593 (shinji), W&B run: `3aaevlho`
- val_primary/surface_rel_l2_pct: **12.70%**
- Model: 4L/256d, T_max=30, AdamW lr=5e-4, no-EMA, Fourier features, 45 epochs (still converging)
- Eval batches: 200

Reproduce baseline:
```bash
cd target/icml2026 && python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 5e-4 \
  --cosine_t_max 30 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200
```

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2652 because ADVISOR: Closing — DrivAerML baseline shifted to 5.73% (4L/384d, PR #2602). All 4L/256d experiments are now obsolete — the best 4L/256d result was 11.97%. Width scaling is the dominant lever. Redirecting your GPU to 4L/384d experiments. You'll receive a new assignment shortly..
